src/plugins/intel_npu/src/backend/src/zero_executor.cpp

```diff
@@ -100,6 +87,51 @@ ZeroExecutor::ZeroExecutor(const std::shared_ptr<const ZeroInitStructsHolder>& i
         }
     }
 
+    if (_graph_ddi_table_ext.version() < ZE_GRAPH_EXT_VERSION_1_8) {
+        initialize_graph_through_command_list();
+    } else {
+        ze_graph_properties_2_t properties = {};
+        properties.stype = ZE_STRUCTURE_TYPE_GRAPH_PROPERTIES;
+        _graph_ddi_table_ext.pfnGetProperties2(_graph, &properties);
+
+        if (properties.initStageRequired & ZE_GRAPH_STAGE_INITIALIZE) {
+            OV_ITT_TASK_NEXT(ZERO_EXECUTOR_GRAPH, "pfnGraphInitialize");
+            _graph_ddi_table_ext.pfnGraphInitialize(_graph);
+        }
+
+        if (properties.initStageRequired & ZE_GRAPH_STAGE_COMMAND_LIST_INITIALIZE) {
```

Comment:
Is it possible to set both `ZE_GRAPH_STAGE_INITIALIZE` and `ZE_GRAPH_STAGE_COMMAND_LIST_INITIALIZE`?

If so, we can potentially call graph initialize twice.

Reply:
It shouldn't be possible right now. But we don't know what the future will be. We should check the bit and if it is set, just call it.

Reply:
If they are mutually exclusive, you can consider one of the into `else if` branch.

But it is more like a nitpick.

Reply:
yeah, I was thinking about this but we don't know for sure in the future how this change. Since that property is represented on bits makes more sense to check the bit instead of doing an if-else. In this way driver should decide if only one is needed, both, or none.